csharp/src/Ice/SignaledSocketStream.cs

```diff
@@ -13,9 +13,27 @@ namespace ZeroC.Ice
     /// for receiving data. The socket can easily signal the stream when new data is available.</summary>
     internal abstract class SignaledSocketStream<T> : SocketStream, IValueTaskSource<T>
     {
-        internal bool IsSignaled => Thread.VolatileRead(ref _signaled) == 1;
-        private volatile Exception? _exception;
-        private int _signaled;
+        internal bool IsSignaled
+        {
+            get
+            {
+                bool lockTaken = false;
+                try
+                {
+                    _lock.Enter(ref lockTaken);
+                    return _source.GetStatus(_source.Version) != ValueTaskSourceStatus.Pending;
+                }
+                finally
+                {
+                    _lock.Exit(lockTaken);
```

Comment:
This does not look correct. It should be:
```
if (lockTaken)
{
    _lock.Exit();
}
```

See https://docs.microsoft.com/en-us/dotnet/api/system.threading.spinlock?view=net-5.0

Reply:
Oops, not sure how I missed that one!

Reply:
Fixed